IsraelHG/solvd-test-automation
# Includes 3 more test cases for mobile automation

## Includes three more mobile automation test cases for the Google Chrome app.

**Test Case 1: testChromeSearch** 
 
This test case verifies the functionality of searching on the Google Chrome application. 

- Launch the Google Chrome application on the mobile device.
- Verify that the ChromeHomePage is opened Successfully. 
- Read the search input from the test data source (mobileGoogleSearches.xlsx). 
- Perform a search on the ChromeHomePage using the provided search query. 
- Verify that the ResultPage is opened successfully, displaying search results. 

 
 
 
**Test Case 2: testChromeTabs** 

This test case verifies the functionality of opening and switching tabs on the Google Chrome application. 

- Launch the Google Chrome application on the mobile device. 
- Verify that the ChromeHomePage is opened successfully. 
- Open a new tab using the ChromeTabsPage. 
- Verify that the ChromeTabsPage is opened successfully, showing the list of tabs. 
- Switch back to the original Google Chrome homepage using the newly created tab. 
- Verify that the original Google Chrome homepage is displayed successfully. 

 

**Test Case 3: testPageInfo** 

This test case verifies the functionality of opening the Chrome information page and checking its content. 

- Launch the Google Chrome application on the mobile device. 
- Verify that the ChromeHomePage is opened successfully. 
- Open the Chrome information page from the menu. 
- Verify that the ChromeInfoPage is opened successfully. 
- Check the information displayed on the ChromeInfoPage and compare it to the expected value. 

 

**Test Case 4: testClearHistory** 

This test case verifies the functionality of clearing browsing history in Google Chrome. 

- Launch the Google Chrome application on the mobile device. 
- Verify that the ChromeHomePage is opened successfully. 
- Perform a search for a specific query on the ChromeHomePage. 
- Verify that the ResultPage is opened successfully. 
- Go back to the ChromeHomePage from the ResultPage. 
- Open the history page from the menu. 
- Verify that the ChromeHistoryPage is opened successfully. 
- Initiate the process of clearing browsing data. 
- Verify that the ClearBrowsingDataPage is opened successfully. 
- Clear the browsing data and go back to the ChromeHistoryPage. 
- Verify that the ChromeHistoryPage is displayed successfully after clearing the history. 

 

**Test Case 5: testBookmarkFeature** 

This test case verifies the functionality of bookmarking a page and deleting the bookmark. 

- Launch the Google Chrome application on the mobile device. 
- Verify that the ChromeHomePage is opened successfully. 
- Perform a search for a specific query on the ChromeHomePage. 
- Verify that the ResultPage is opened successfully. 
- Open the menu from the ResultPage and add the current page to bookmarks. 
- Verify that the ChromeBookmarkPage is opened successfully. 
- Open the mobile bookmarks page from the ChromeBookmarkPage. 
- Verify that the ChromeBookmarkMobileListPage is opened successfully. 
- Delete the bookmark using the options menu on the ChromeBookmarkMobileListPage. 
- Verify that the ChromeBookmarkPage is displayed successfully after deleting the bookmark.
